### PR TITLE
fix(dropdown): avoid invalid dropdown icons overlapping with option text

### LINKED ISSUE #1897: Invalid dropdown icons overlapping with option text
Invalid dropdowns are not accounting for the extra padding needed for the additional SVG icon in order to display the full option text

image

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1897

#### Changelog

**Changed**

- increase right padding of invalid state dropdown options
- avoid data attribute selectors

### CODE REVIEW

File: src/components/dropdown/_dropdown.scss [hunk 1]
```diff
@@ -256,9 +256,13 @@
     }
   }
 
-  .#{$prefix}--dropdown[data-invalid] {
+  .#{$prefix}--dropdown--invalid {
```
Comment (asudoh) [inline src/components/dropdown/_dropdown.scss:259]: Does this change cover React dropdown (`<DropdownV2>`) as well as vanilla dropdown?

Comment (emyarod) [reply src/components/dropdown/_dropdown.scss:259]: from what I can tell, the React dropdown does not account for invalid state, so this change won't affect it (and the React component needs to be extended...we can create a new ticket in the React repo for this). please let me know if this is incorrect

Comment (asudoh) [reply src/components/dropdown/_dropdown.scss:259]: Oh sorry I had a false assumption that `--invalid` is for React. Makes sense now.

Comment (asudoh) [reply src/components/dropdown/_dropdown.scss:259]: Could you update `migrate-to-10.x.md` mentioning the change in how to apply the invalid style? Thanks!

Comment (emyarod) [reply src/components/dropdown/_dropdown.scss:259]: sure I updated that document

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Review by tw15egan — APPROVED** [review]
LGTM 👍

**Pull request merged**